### PR TITLE
chore(core): remove deprecated uses of can_write_to_container()

### PR DESCRIPTION
**Pull request by mrclay**
Also no longer used `Request::stripSlashesIfMagicQuotes`
- ~~[ ] waiting on #10023~~ Separate issue.

### CODE REVIEW

File: mod/blog/actions/blog/save.php [hunk 1]
```diff
@@ -91,7 +91,8 @@
 		case 'container_guid':
 			// this can't be empty or saving the base entity fails
 			if (!empty($value)) {
-				if (can_write_to_container($user->getGUID(), $value)) {
+				$container = get_entity($value);
+				if ($container && $container->canWriteToContainer()) {
```
Comment (hypeJunction) [inline mod/blog/actions/blog/save.php]: Could we start requesting subtype-specific permissions when we can? 
`$container->canWriteToContainer(0, 'object', 'blog')`

**Comment by hypeJunction** [comment]
I am quite sure I have updated these at some point when working on UserCapabilities. I will dig into my old PR.

**Comment by mrclay** [comment]
I made the canWriteToContainer() calls more specific, but is there a BC issue to consider?

**Comment by hypeJunction** [comment]
Hm, BC is a good point, but if we were requesting generic `canWriteToContainer` permissions to display a form, `$entity->save()` would eventually fail anyway. So, IMO it's a bug fix.

File: mod/pages/start.php [hunk 2]
```diff
@@ -370,8 +370,8 @@ function pages_container_permission_check($hook, $entity_type, $returnvalue, $pa
 	if (elgg_get_context() != "pages") {
```
Comment (hypeJunction) [inline mod/pages/start.php]: This could die.

Comment (mrclay) [inline mod/pages/start.php]: It's not necessary at all?

**Comment by hypeJunction** [comment]
Pages are completely broken with regards to permissions. Sending you a PR.

**Comment by mrclay** [comment]
> Pages are completely broken with regards to permissions

If you mean on the 2.x branch, please make a new ticket so we're all aware.

**Comment by hypeJunction** [comment]
I think they were broken since 2.0, but I don't care enough to look.

**Comment by hypeJunction** [comment]
#10023

File: mod/pages/start.php (hunk 2 above)
Comment (mrclay) [inline mod/pages/start.php]: Agreed, not necessary

**Pull request merged**